doc/source/docs/user_guide/sampling.ipynb

```diff
@@ -0,0 +1,268 @@
+    "To specify the sample sizes, provide an explicit number of points to sample. For example, we can sample 200 points randomly from each feature: "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "76d8658d-7ee8-4ef0-84b4-b8883c921687",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "n200_sampled_points = nybb.sample_points(200)\n",
+    "m = nybb.explore()\n",
+    "n200_sampled_points.explore(m=m, color='red')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "14b8628f-3d5e-4fbb-b26e-28cc042ff755",
+   "metadata": {},
+   "source": [
+    "This functionality also works for line geometries. For example, let's look only at the boundary of Manhattan Island:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c47472fc-7ca5-4f69-b86f-93c25a9f2b03",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "manhattan_parts = nybb.iloc[[3]].explode(ignore_index=True)\n",
+    "manhattan_island = manhattan_parts.iloc[[30]]\n",
+    "manhattan_island.boundary.explore()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "54280acd-e449-4528-ac7c-c1b2dbbcdc2f",
+   "metadata": {},
+   "source": [
+    "Sampling randomly from along this boundary can use the same `sample_points()` method:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "358822ef-a0c6-40cd-9e35-9a688c56f361",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "manhattan_border_points = manhattan_island.boundary.sample_points(200)\n",
+    "m = manhattan_island.explore()\n",
+    "manhattan_border_points.explore(m=m, color='red')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fa0f5a05-e7cc-44cf-b6ac-40afd9753f23",
+   "metadata": {},
+   "source": [
+    "Keep in mind that sampled points are returned as a single multi-part geometry, and that the distances over the line segments are calculated *along* the line. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b0ccad29-d29f-4da1-9151-692bfd20d533",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "manhattan_border_points"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "125f19d9-b3f0-4ab9-b44a-41be1a8cc388",
+   "metadata": {},
+   "source": [
+    "If you want to separate out the individual sampled points, use the `.explode()` method on the dataframe:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "265f2194-a94f-4a3d-9ae8-f7da55893e90",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "manhattan_border_points.explode(ignore_index=True).head()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "21a1a9d5",
+   "metadata": {},
+   "source": [
+    "## Variable number of points\n",
+    "\n",
+    "You can also sample different number of points from different geometries if you pass an array specifying the size of the sample per geometry."
```

Comment:
Do we allow to pass a column name as well? (instead of just the values, so I assume `gdf.sample_points(gdf["col"])` works for sure)

Reply:
As we have in plotting? Not now. Do we want to?